single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -468,8 +535,23 @@ def new_leader(self) -> None:
         if not self.state.get_keyfile():
             self.state.set_keyfile(self.workload.generate_keyfile())
 
-        # Sets the password for the system users
-        for user in (OperatorUser, BackupUser, MonitorUser, LogRotateUser):
+        if self.state.internal_user_passwords_are_initialized():
+            return
+
+        if system_users_secret_id := self.config.system_users:
+            user_passwords = None
+            try:
+                user_passwords = self.charm.state.get_secret_from_id(system_users_secret_id)
+                validate_charm_user_password_config(user_passwords)
+            except (ModelError, SecretNotFoundError, InvalidPasswordError) as e:
+                logger.error(f"Invalid system-users secret: {e}.")
```

Comment:
I feel this should lead to the same kind of message as on config changed: similar problems lead to similar statuses.
It could hence be exported to a helper function that could be used as well in the `get_statuses` method to compute the same statuses on status update events.